agenda/data/src/main/java/com/gkp/agenda/data/image/AndroidImageReader.kt

```diff
@@ -0,0 +1,29 @@
+package com.gkp.agenda.data.image
+
+import android.content.Context
+import android.net.Uri
+import com.gkp.agenda.domain.image.ImageInfo
+import com.gkp.agenda.domain.image.ImageReader
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+
+class AndroidImageReader(
+    private val context: Context,
+    private val dispatcher: CoroutineDispatcher = Dispatchers.IO,
+) : ImageReader {
+    override suspend fun readImagesFromStringUri(
+        stringUris: List<String>,
+    ): List<ImageInfo> {
+        return withContext(dispatcher) {
+            stringUris.map { uriString ->
+                val content = context.contentResolver.openInputStream(Uri.parse(uriString))?.use {
+                    it.readBytes()
+                } ?: byteArrayOf()
+                val type = context.contentResolver.getType(Uri.parse(uriString)) ?: ""
+                val compressedImage = ImageCompressor.compressImage(content)
+                ImageInfo(type, uriString, compressedImage)
+            }
```

Comment:
This function could take a while to complete, depending on the size of the images. This is because the map function is synchronous and will only map the next element in the list once the first one has finished. 

You can change this code to be asynchronous by using `async { }`

```
override suspend fun readImagesFromStringUri(
  stringUris: List<String>,
): List<ImageInfo> {
  val deferredResults = withContext(dispatcher) {
    stringUris.map { uriString ->
      async {
        // process content here and return ImageInfo
      }
    }
  return deferredResults.awaitAll()
}
```